dbt/adapters/bigquery/retry.py

```diff
@@ -1,10 +1,9 @@
 from typing import Callable, Optional
 
-from google.api_core.exceptions import Forbidden
 from google.api_core.future.polling import DEFAULT_POLLING
 from google.api_core.retry import Retry
-from google.cloud.bigquery.retry import DEFAULT_RETRY
-from google.cloud.exceptions import BadGateway, BadRequest, ServerError
+from google.cloud.bigquery.retry import DEFAULT_JOB_RETRY
```

Comment:
In general, we're retrying jobs, we should use this default instead of `DEFAULT_RETRY`. We still use `DEFAULT_RETRY` for the client factories where it makes more sense.